assets/src/editor/events/ticket-assignments-manager/hooks/use-counts-manager.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External imports
+ */
+import { useMemo, useCallback } from '@wordpress/element';
+import { reduce, cloneDeep, isEmpty } from 'lodash';
+import { __ } from '@eventespresso/i18n';
+
+// @todo, when the wp.data.useSelect hook is available, we can utilize that in
+// this custom hook and then we'll be able to remove the withSelect hoc usage
+// in the composed `TicketAssignmentsManagerModal` component.
+
+const useCountsManager = (
+	dateEntities,
+	ticketEntities,
+	assignedCounts,
+	assignedState,
+) => {
+	const getMessage = useCallback( ( whichAreEmpty ) => {
+		let message;
+		const onlyTicketsAreEmpty = whichAreEmpty.dates === 0 && whichAreEmpty.tickets > 0;
+		const onlyDatesAreEmpty = whichAreEmpty.dates > 0 && whichAreEmpty.tickets === 0;
+		const anyAreEmpty = whichAreEmpty.dates > 0 || whichAreEmpty.tickets > 0;
+
+		switch ( true ) {
+			case ( dateEntities.length === 1 && onlyDatesAreEmpty ) ||
+			( ticketEntities.length === 1 && onlyDatesAreEmpty ) :
+				message = __(
+					'Event Dates must always have at least one Ticket assigned to them. If the current assignment is not correct, assign the correct Ticket first, then remove others as required.',
+					'event_espresso'
+				);
+				break;
+			case ( ticketEntities.length === 1 && anyAreEmpty ) ||
+			( dateEntities.length === 1 && onlyTicketsAreEmpty ) :
+				message = __(
+					'Tickets must always have at least one Event Date assigned to them. If the current assignment is not correct, assign the correct Event Date first, then remove others as required.',
+					'event_espresso'
+				);
+				break;
+			default:
+				message = __(
+					'Event Dates must always have at least one Ticket assigned to them but one or more of the Event Dates below does not have any. Please correct the assignments for the highlighted cells.',
+					'event_espresso'
+				);
+		}
+		return message;
+	}, [ dateEntities, ticketEntities ] );
+
+	const isEmptyAssignedState = useCallback(
+		() => isEmpty( assignedState.assigned ) && isEmpty( assignedState.removed ),
+		[ assignedState ]
+	);
+
+	const [ hasNoAssignments, noAssignmentsMessage, updatedAssignmentsCounts ] = useMemo( () => {
```

Comment:
Instead of `updatedAssignmentsCounts` being a ref, it is now contained in the `useMemo` implementation and returned as a value.  Much less fragile.